### PR TITLE
[RLlib] Test cases/BUILD cleanup; split "everything else" (longest running one rn) tests in 2.

### PR DESCRIPTION
**Pull request by sven1977**
Test cases/BUILD cleanup; split "everything else" (longest running one rn) tests in 2: "trainers_dir" and "everything else".
- Fixes a bug in async PG-style algos (A3C and APPO) for LSTM/RNN models.
- Remove more of the soon-to-be-deprecated quick_train tests.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/a3c/tests/test_a3c.py [hunk 1]
```diff
@@ -25,13 +25,15 @@ def test_a3c_compilation(self):
 
         # Test against all frameworks.
         for _ in framework_iterator(config):
-            for env in ["CartPole-v0", "Pendulum-v0", "PongDeterministic-v0"]:
+            for env in ["CartPole-v1", "Pendulum-v0", "PongDeterministic-v0"]:
```
Comment (michaelzhiluo) [inline rllib/agents/a3c/tests/test_a3c.py:28]: V1 Cartpole goes to 500 reward, not 200, but change the results fyi

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -252,7 +252,19 @@
       --test_env=RAY_USE_MULTIPROCESSING_CPU_COUNT=1
       rllib/...
 
-- label: ":brain: RLlib: everything else"
+- label: ":brain: RLlib: Trainer Tests"
```
Comment (michaelzhiluo) [inline .buildkite/pipeline.yml:255]: What is the intended change here?

File: rllib/agents/es/tests/test_es.py [hunk 3]
```diff
@@ -25,14 +25,15 @@ def test_es_compilation(self):
...
         for _ in framework_iterator(config):
-            plain_config = config.copy()
-            trainer = es.ESTrainer(config=plain_config, env="CartPole-v0")
-            for i in range(num_iterations):
-                results = trainer.train()
-                print(results)
-
-            check_compute_single_action(trainer)
-            trainer.stop()
+            for env in ["CartPole-v0", "Pendulum-v0"]:
```
Comment (michaelzhiluo) [inline rllib/agents/es/tests/test_es.py:28]: This can time out right?

File: rllib/agents/pg/tests/test_pg.py [hunk 4]
```diff
@@ -22,11 +22,12 @@ def tearDown(self):
...
         config = pg.DEFAULT_CONFIG.copy()
-        config["num_workers"] = 0
-        num_iterations = 2
+        config["num_workers"] = 1
+        config["rollout_fragment_length"] = 500
+        num_iterations = 1
 
         for _ in framework_iterator(config):
-            trainer = pg.PGTrainer(config=config, env="CartPole-v0")
+            trainer = pg.PGTrainer(config=config, env="FrozenLake-v0")
```
Comment (michaelzhiluo) [inline rllib/agents/pg/tests/test_pg.py:30]: Maybe loop through both CartPole and FrozenLake as envs

**Review by michaelzhiluo — APPROVED** [review]

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (sven1977) [reply .buildkite/pipeline.yml:255]: To separate the trainer compilation tests (basically all tests that are in the `agents` dir) from the other tests in this suite (e.g. models, policy, utils, etc..).

File: rllib/agents/es/tests/test_es.py (hunk 3 above)
Comment (sven1977) [reply rllib/agents/es/tests/test_es.py:28]: We only run for a single iteration, no learning testing here. Should be fine.

File: rllib/agents/pg/tests/test_pg.py (hunk 4 above)
Comment (sven1977) [reply rllib/agents/pg/tests/test_pg.py:30]: +1

Comment (sven1977) [reply rllib/agents/pg/tests/test_pg.py:30]: done

File: rllib/agents/a3c/tests/test_a3c.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/a3c/tests/test_a3c.py:28]: That's ok, it's just a "compilation" test. No learning progress checked here.
